modules/dxp/apps/search-experiences/search-experiences-service/src/main/java/com/liferay/search/experiences/internal/blueprint/search/spi/searcher/SXPBlueprintSearchRequestContributor.java

```diff
@@ -48,12 +49,43 @@ public SearchRequest contribute(SearchRequest searchRequest) {
 		SearchRequestBuilder searchRequestBuilder =
 			_searchRequestBuilderFactory.builder(searchRequest);
 
+		_contributeSXPBlueprintExternalReferenceCode(searchRequestBuilder);
 		_contributeSXPBlueprintId(searchRequestBuilder);
 		_contributeSXPBlueprintJSON(searchRequestBuilder);
 
 		return searchRequestBuilder.build();
 	}
 
+	private void _contributeSXPBlueprintExternalReferenceCode(
+		SearchRequestBuilder searchRequestBuilder) {
+
+		Object object = searchRequestBuilder.withSearchContextGet(
+			searchContext -> searchContext.getAttribute(
+				"search.experiences.blueprint.erc"));
+
+		if (_log.isDebugEnabled()) {
+			_log.debug(
+				"Search experiences blueprint External Reference Code " +
+					object);
+		}
+
+		if (object == null) {
```

Comment:
Maybe removing this void check and simplify by replacing the last else on line 82 with `else if (object != null)`

The same for the existing  _contributeSXPBlueprintId()